sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -294,22 +293,18 @@ public void withScope(@NotNull ScopeCallback callback) {
           "Instance is disabled and this 'withScope' call is a no-op.");
     } else {
       pushScope();
-      try {
-        StackItem item = stack.peek();
-        if (item != null) {
-          try {
-            callback.run(item.scope);
-          } catch (Exception e) {
-            logIfNotNull(
-                options.getLogger(), SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
-          }
-        } else {
+      StackItem item = stack.peek();
+      if (item != null) {
+        try {
+          callback.run(item.scope);
+        } catch (Exception e) {
           logIfNotNull(
-              options.getLogger(), SentryLevel.FATAL, "Stack peek was null when withScope");
+              options.getLogger(), SentryLevel.ERROR, "Error in the 'withScope' callback.", e);
         }
-      } finally {
-        popScope();
+      } else {
+        logIfNotNull(options.getLogger(), SentryLevel.FATAL, "Stack peek was null when withScope");
```

Comment:
I am seeing a lot of calls to `logIfNotNull` using `options.getLogger()` as the `ILogger` instance, which is marked as `@NotNull` and safely handled as a non-null value in `SentryOptions`.
What is the purpose of checking for nullability in the log call then?

Reply:
@ninniuz so basically this was supposed to be like an extension method, and I guess that time `logger` could be `null`.
I'd say we need to revisit the whole `log` thing to work as `.net`, a single `log` call that checks nullability, if it could be null, if `debug` is enabled and also if the default `logLevel` or the one set by the client is >= the given `logLevel`, we have this task already and it's a good point.